search/routes/api/atom_extensions.py

```diff
@@ -0,0 +1,293 @@
+from feedgen.ext.base import BaseEntryExtension, BaseExtension
+from feedgen.entry import FeedEntry
+from feedgen.feed import FeedGenerator
+from lxml import etree
+
+ARXIV_NS = 'http://arxiv.org/schemas/atom'
+OPENSEARCH_NS = 'http://a9.com/-/spec/opensearch/1.1/'
+
+class OpenSearchExtension(BaseExtension):
+    """Extension of the Feedgen base class to put OpenSearch metadata."""
+
+    # pylint: disable=invalid-name
+
+    def __init__(self: BaseExtension) -> None:
+        """Initialize extension parameters."""
+        # __ syntax follows convention of :module:`feedgen.ext`
+        self.__opensearch_totalResults = None
+        self.__opensearch_startIndex = None
+        self.__opensearch_itemsPerPage = None
+
+    def extend_atom(self: BaseExtension, atom_feed: FeedGenerator) -> FeedGenerator:
+        """
+        Assign the Atom feed generator to the extension.
+
+        Parameters
+        ----------
+        atom_feed : :class:`.FeedGenerator`
+            The FeedGenerator to use for Atom results.
+
+        Returns
+        -------
+        FeedGenerator
+            The provided feed generator.
+
+        """
+        if self.__opensearch_itemsPerPage is not None:
+            elt = etree.SubElement(atom_feed, f'{{{OPENSEARCH_NS}}}itemsPerPage')
+            elt.text = self.__opensearch_itemsPerPage
+
+        if self.__opensearch_totalResults is not None:
+            elt = etree.SubElement(atom_feed, f'{{{OPENSEARCH_NS}}}totalResults')
+            elt.text = self.__opensearch_totalResults
+
+        if self.__opensearch_startIndex is not None:
+            elt = etree.SubElement(atom_feed, f'{{{OPENSEARCH_NS}}}startIndex')
+            elt.text = self.__opensearch_startIndex
+
+        return atom_feed
+
+    @staticmethod
+    def extend_rss(rss_feed: FeedGenerator) -> FeedGenerator:
+        """
+        Assign the RSS feed generator to the extension.
+
+        Parameters
+        ----------
+        rss_feed
+            The FeedGenerator to use for RSS results.
+
+        Returns
+        -------
+        FeedGenerator
+            The provided feed generator.
+
+        """
+        return rss_feed
+
+    @staticmethod
+    def extend_ns() -> Dict[str, str]:
+        """
+        Assign the feed's namespace string.
+
+        Returns
+        -------
+        str
+            The definition string for the "arxiv" namespace.
+
+        """
+        return {'opensearch': OPENSEARCH_NS}
+
+    def totalResults(self: BaseExtension, text: str) -> None:
+        """Set the totalResults parameter."""
+        self.__opensearch_totalResults = str(text)
+
+    def startIndex(self: BaseExtension, text: str) -> None:
+        """Set the startIndex parameter."""
+        self.__opensearch_startIndex = str(text)
+
+    def itemsPerPage(self: BaseExtension, text: str) -> None:
+        """Set the itemsPerPage parameter."""
+        self.__opensearch_itemsPerPage = str(text)
+
+
+class ArxivExtension(BaseExtension):
```

Comment:
This is super nit-picky, but can we upper the x?
```suggestion
class ArXivExtension(BaseExtension):
```